### DISCUSSION TITLE
Convert a notebook from cli

### DISCUSSION THREAD

**Discussion by mil-ad** [datapane/datapane#349] [Help]
I'm trying to turn my dp report generation into a script and was wondering if there's a way to directly convert a jupyter notebook (i.e. equivalent of `dp.App(blocks=dp.cells_to_blocks()).save(path="report.html")`) to a dp App from the command line?

`datapane app --help` doesn't tell me enough.

**Comment by spookylukey** [comment]
There isn't any way to do this yet. From what I can see, this wouldn't be a trivial feature addition, because of the way that `cells_to_blocks` relies on the running notebook. However, I'm pretty sure this is something we would want to support.

Another way to approach this is to run your notebook with something like nbconvert, especially the execute feature:

https://nbconvert.readthedocs.io/en/latest/execute_api.html

I've tried this, and this also doesn't work with datapane 0.15.7 because of the way that the environment detection works. This is definitely we something we should fix as well.
